src/Extensions/AddToCampaignExtension.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace SilverStripe\CampaignAdmin\Extensions;
+
+use RuntimeException;
+use SilverStripe\AssetAdmin\Controller\AssetAdmin;
+use SilverStripe\CampaignAdmin\AddToCampaignHandler;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\ArrayLib;
+use SilverStripe\Core\Extension;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\Form;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Extension applied to a Controller to add the ability to add items to a campaign.
+ * This extension will only be applied if the `campaign-admin` module is installed.
+ *
+ * @extends Extension<AssetAdmin>
```

Comment:
```suggestion
 * @extends Extension<AssetAdmin|CMSPageEditController>
```
Or could it theoretically be added to any `LeftAndMain`?

Reply:
Updated to LeftAndMain